python/dataset.cpp

```diff
@@ -253,22 +253,25 @@ void init_dataset(py::module &m) {
   py::class_<DataArray> dataArray(m, "DataArray", R"(
     Named variable with associated coords, masks, and attributes.)");
   dataArray.def(py::init<const DataArrayConstView &>());
-  dataArray.def(
-      py::init([](VariableConstView data,
-                  std::map<Dim, VariableConstView> coords,
-                  std::map<std::string, VariableConstView> masks,
-                  std::map<Dim, VariableConstView> unaligned_coords,
-                  const std::string &name) {
-        return DataArray{Variable{data}, coords, masks, unaligned_coords, name};
-      }),
-      py::arg("data") = Variable{},
-      py::arg("coords") = std::map<Dim, VariableConstView>{},
-      py::arg("masks") = std::map<std::string, VariableConstView>{},
-      py::arg("unaligned_coords") = std::map<Dim, VariableConstView>{},
-      py::arg("name") = std::string{});
+  dataArray
+      .def(py::init([](VariableConstView data,
+                       std::map<Dim, VariableConstView> coords,
+                       std::map<std::string, VariableConstView> masks,
+                       std::map<Dim, VariableConstView> unaligned_coords,
+                       const std::string &name) {
+             return DataArray{Variable{data}, coords, masks, unaligned_coords,
+                              name};
+           }),
+           py::arg("data") = Variable{},
+           py::arg("coords") = std::map<Dim, VariableConstView>{},
+           py::arg("masks") = std::map<std::string, VariableConstView>{},
+           py::arg("unaligned_coords") = std::map<Dim, VariableConstView>{},
+           py::arg("name") = std::string{})
+      .def("__sizeof__", &DataArray::sizeInMemory);
 
   py::class_<DataArrayConstView>(m, "DataArrayConstView")
-      .def(py::init<const DataArray &>());
+      .def(py::init<const DataArray &>())
+      .def("__sizeof__", &DataArrayConstView::sizeInMemory);
```

Comment:
Any reason for not naming the C++ function closer to the Python equivalent, e.g., `size_of`?